openshift-knative-operator/pkg/serving/extension_test.go

```diff
@@ -113,27 +125,61 @@ func TestReconcile(t *testing.T) {
 		expected: ks(func(ks *v1alpha1.KnativeServing) {
 			ks.Status.MarkDependenciesInstalled()
 		}),
+		shouldDisableMonitoring: false,
+	}, {
+		name: "respect already configured metrics backend",
+		in: &v1alpha1.KnativeServing{
+			Spec: v1alpha1.KnativeServingSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					Config: map[string]map[string]string{monitoring.ObservabilityCMName: {monitoring.ObservabilityBackendKey: "prometheus"}},
+				},
+			},
+		},
+		expected: ks(func(ks *v1alpha1.KnativeServing) {
+			common.Configure(&ks.Spec.CommonSpec, monitoring.ObservabilityCMName, monitoring.ObservabilityBackendKey, "prometheus")
+		}),
+		shouldDisableMonitoring: false,
+	}, {
+		name: "disable monitoring by default",
+		in:   &v1alpha1.KnativeServing{},
+		expected: ks(func(ks *v1alpha1.KnativeServing) {
+			common.Configure(&ks.Spec.CommonSpec, monitoring.ObservabilityCMName, monitoring.ObservabilityBackendKey, "none")
+		}),
+		shouldDisableMonitoring: true,
 	}}
-
+	ctx, _ := injection.EnableInjectionOrDie(context.Background(), nil)
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
 			objs := c.objs
 			if objs == nil {
 				objs = []runtime.Object{defaultIngress}
 			}
-			ctx, _ := ocpfake.With(context.Background(), objs...)
-
 			ks := c.in.DeepCopy()
-			ext := NewExtension(ctx)
+			ks.Namespace = "knative-serving"
+			c.expected.Namespace = ks.Namespace
+			ctx, _ = ocpfake.With(ctx, objs...)
+			ext := NewExtension(ctx).(*extension)
+			ext.kubeclient = fakekubeclientset.NewSimpleClientset([]runtime.Object{&servingNamespace}...)
```

Comment:
Let's not construct this manually. You can use the same `With` function as is used for the `ocpfake` above. That also avoids the unnecessary cast to `*extension`

Reply:
@markusthoemmes the returned clientSet from ocpfake.With call, only provides access to:
```
	Discovery() discovery.DiscoveryInterface
	ConfigV1() configv1.ConfigV1Interface
	RouteV1() routev1.RouteV1Interface
```
I cannot use it directly to get the namespace bellow that is why I used this fake client explicitly. What do you suggest exactly?

Reply:
You can import the same fake client for K8s and do `kubefake.With(ctx, &servingNamespace)`

Reply:
Not sure I see that method I just followed what I read [here](https://github.com/knative/pkg/blob/448ae657fb5fb0d84d99825d184d706478574023/metrics/reflector_test.go#L62).